### ISSUE TITLE
2fa qr code

### ISSUE THREAD

**Issue by tomastaker** [3urobeat/steam-comment-service-bot#193]
Hello, is it possible to implement authorization 2fa accounts with a QR code, as this did here? 
https://github.com/JakyeRU/SteamIdler

This method allows you to connect to accounts using only a .env file, without logins and passwords

upd 
As I saw, the env code generated by the application that I pointed out above also works with your application, however, I do not understand 1 detail

What does this part in tokens.db mean
"_id":"bK6jNXaqlL5hVvV5"
How do I get this account id?

**Comment by tomastaker** [comment]
> Hello, is it possible to implement authorization 2fa accounts with a QR code, as this did here? https://github.com/JakyeRU/SteamIdler
> 
> This method allows you to connect to accounts using only a .env file, without logins and passwords
> 
> upd As I saw, the env code generated by the application that I pointed out above also works with your application, however, I do not understand 1 detail
> 
> What does this part in tokens.db mean "_id":"bK6jNXaqlL5hVvV5" How do I get this account id?

UPD
Surprisingly, it all worked natively, just by transferring the ENV key))

It remains only to understand how to unite the functionality to accounts (bots) from which I send comments, and also run the game on the hour boost 

You have an amazing project)

**Comment by 3urobeat** [comment]
Hey, thanks for opening an issue!  
I have considered adding QR-Code support in the past but didn't commit to it for some reason I can't remember. I should revisit the feature however, thanks for suggesting it.   

I'm not sure what the `.env` file you mentioned is, I'd have to look that up.  

The `_id` parameter you are seeing in the database is an internal parameter of the database which me must not change, it is used to uniquely identify each document. This is explained [here](https://github.com/seald/nedb/#inserting-documents).  
I'm only using the `accountName` to associate a `refreshToken`.

Thanks for the kind words!

You don't need the hour boost you mentioned, this bot can idle games for you just fine. You also can't use both at the same time because you can only have one active SteamCommunity session at a time.  
Check out the `playinggames` and `childaccplayinggames` settings in the `config.json`.  
You can read how to use them in the [config documentation](https://github.com/3urobeat/steam-comment-service-bot/blob/master/docs/wiki/config_doc.md).  

In short, `playinggames` sets the game which the first account (which you are chatting with) plays. The `childaccplayinggames` setting sets the game of all other accounts (called child accounts).  
You can provide a custom game as the first parameter (just like in the example) and all other params will be game ids.

I hope this helps you :)

**Comment by tomastaker** [comment]
Wow, you made just an incredible combine of features))) 

Could you tell me, is it possible to set unique games for each child account? 

For example 
Account 1 [720, 570]
account 2 [570, 440, 671860]

**Comment by 3urobeat** [comment]
Well, commenting has become one of the smaller parts of the bot tbh  

Idling different games for each account is not possible yet. If you however only want to idle certain games because the other accounts do not own them then that shouldn't be a problem, they just won't play it even though it's set.

**Comment by 3urobeat** [comment]
I have added support for setting specific games per account. This feature is ported over from my steam-idler and uses the same settings but documentation will follow in this project soon.

**Comment by 3urobeat** [comment]
Version 2.15 is now out on the master branch and adds support for logging in using QR-Codes.  
Simply provide the password `qrcode` for an account in `accounts.txt` to use this feature.  
Check out the release notes [here](https://github.com/3urobeat/steam-comment-service-bot/releases/tag/2.15.0)!
